adr/0004-using-the-tls-registry-for-clients.adoc

```diff
@@ -0,0 +1,192 @@
+
+Clients must not use the default TLS configuration, except for one case.
+For backward compatibility purpose, `quarkus.tls.trust-all` should be honored when configuring clients.
+
+== Using the TLS registry from client extensions
+
+=== Retrieving the registry from an extension
+
+The TLS registry is a CDI bean and can be injected in other beans.
+The TLS registry is a _singleton_ CDI bean.
+
+However, before configuring the client, you need to make sure the TLS registry is properly initialized.
+The recommended approach is to inject the `TlsConfigurationRegistry tlsConfigurationRegistry` bean in a `recorder`:
+
+[source, java]
+----
+BasicWebSocketConnectorImpl(Vertx vertx, Codecs codecs, ClientConnectionManager connectionManager,
+     WebSocketsClientRuntimeConfig config, TlsConfigurationRegistry tlsConfigurationRegistry) {
+    // ...
+}
+----
+
+Another approach is to use the `TlsRegistryBuildItem` in a _processor_:
+
+[source, java]
+----
+@Record(ExecutionTime.RUNTIME_INIT)
+@BuildStep
+public SyntheticBeanBuildItem setup(OidcBuildTimeConfig oidcBuildTimeConfig, OidcConfig oidcRunTimeConfig,
+        KeycloakPolicyEnforcerConfig keycloakConfig, KeycloakPolicyEnforcerRecorder recorder,
+        HttpConfiguration httpConfiguration, TlsRegistryBuildItem tlsRegistryBuildItem) {
+        if (oidcBuildTimeConfig.enabled) {
+            return SyntheticBeanBuildItem.configure(PolicyEnforcerResolver.class).unremovable()
+                    .types(PolicyEnforcerResolver.class)
+                    .supplier(recorder.setup(oidcRunTimeConfig, keycloakConfig, httpConfiguration,
+                            tlsRegistryBuildItem.registry())) // Get the registry runtime value
+                    .scope(Singleton.class)
+                    .setRuntimeInit()
+                    .done();
+        }
+        return null;
+}
+----
+
+=== Finding the named TLS configuration
+
+Once the TLS registry is properly passed, the client extension can use it to retrieve the named TLS configuration:
+
+[source, java]
+----
+TlsConfiguration configuration = null;
+if (config.tlsConfigurationName.isPresent()) {
+    Optional<TlsConfiguration> maybeConfiguration = tlsRegistry.get(config.tlsConfigurationName.get());
+    if (!maybeConfiguration.isPresent()) {
+        // We should reject missing named configuration
+        throw new IllegalStateException("Unable to find the TLS configuration "
+                + config.tlsConfigurationName.get() + " for the <enter client name> " + name + ".");
+    }
+    configuration = maybeConfiguration.get();
+} else if (tlsRegistry.getDefault().isPresent() && tlsRegistry.getDefault().get().isTrustAll()) {
+    defaultTrustAll = tlsRegistry.getDefault().get().isTrustAll();
+    if (defaultTrustAll) {
+        LOGGER.warn("The default TLS configuration is set to trust all certificates. This is a security risk."
+                + "Please use a named TLS configuration for the mailer " + name + " to avoid this warning.");
+    }
+}
```

Comment:
We have `TlsConfiguration#from` that does the work of obtaining the proper `TlsConfiguration`

Reply:
Fixed.